networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1220 @@
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+
+    Notes
+    ------
+    WIP TODO: As multiple independent graphs may form from applying threshold cutoffs to a weighted graph,
+    only the largest is considered. This might be worth considering in re-evaluating the meaning of
+    weighted network hierarchy coordinate evaluations. (See pages 7, 8 of [1]_, supplementary material)
+
+    An threshold_distribution of None results in a linear distribution, otherwise
+     the exponential distribution is sampled from exp(x) \in (0, 1)
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+    """
+    import numpy as np
+
+    # binary check
+    if np.array_equal(np.unique(A), [0, 1]):
+        num_thresholds = 1
+    else:
+        num_thresholds = num_thresholds
+
+    # Establishing Thresholds
+    if num_thresholds == 1 or np.isclose(np.max(A) - np.min(A), 0, 1e-2):
+        nx_graphs = [nx.from_numpy_matrix(A, create_using=nx.DiGraph)]
+    else:
+        if threshold_distribution is None:
+            try:
+                thresholds = list(
+                    np.round(
+                        np.arange(
+                            np.min(A),
+                            np.max(A),
+                            (np.max(A - np.min(A))) / num_thresholds,
+                        ),
+                        4,
+                    )
+                )  # linear distribution
+            except:
+                thresholds = [np.max(A)] * num_thresholds
+        else:
+            thresholds = _distribute(
+                dist=threshold_distribution,
+                end_value_range=(np.min(A), np.max(A)),
+                n=num_thresholds,
+            )
+        # Converting to binary nx_graphs according to thresholds:
+        nx_graphs = [
+            nx.from_numpy_matrix(np.where(A > threshold, 1, 0), create_using=nx.DiGraph)
+            for threshold in thresholds
+        ]
+    # removes isolated nodes (0 in & out degree) from binary nodes. (not needed for consolidation)
+    for G in nx_graphs:
+        G.remove_nodes_from(list(nx.isolates(G)))
+
+    # eliminates empty graphs
+    nx_graphs = [graph for graph in nx_graphs if not nx.is_empty(graph)]
+    condensed_graphs = [nx.condensation(G) for G in nx_graphs]
+    largest_condensed_graphs = []
+    for condensed_graph in condensed_graphs:
+        largest_condensed_graphs.append(
+            nx.convert_node_labels_to_integers(
+                max(
+                    [
+                        condensed_graph.subgraph(cc).copy()
+                        for cc in nx.weakly_connected_components(condensed_graph)
+                    ],
+                    key=len,
+                )
```

Comment:
Can the `subgraph(cc).copy()` be pulled out of the `max`? I think `max(nx.weakly_...), key=len)` will give the largest `cc`.

Reply:
Just comparing the two, it appears `max(nx.weakly_...), key=len) ` yields only the set of nodes in the maxima. (Class `Sized`)
```
print(max([condensed_graph.subgraph(cc).copy() for cc in nx.weakly_connected_components(condensed_graph)], key=len))
print(max(nx.weakly_connected_components(condensed_graph), key=len))
```
So in short, it's likely possible, though I know not how.

Reply:
```python
max_cc = max(nx.weakly_connected_components(condensed_graph), key=len))
condensed_graph.subgraph(max_cc).copy()
```
I guess it could all be on one line too if you prefer.